src/node_file.cc

```diff
@@ -3186,7 +3186,9 @@ static void CpSyncCheckPaths(const FunctionCallbackInfo<Value>& args) {
 
   std::string dest_path_str = dest_path.string();
   // Check if dest_path is a subdirectory of src_path.
-  if (src_is_dir && dest_path_str.starts_with(src_path.string())) {
+  if (src_is_dir &&
+      dest_path_str.starts_with(src_path.string() +
+                                std::filesystem::path::preferred_separator)) {
```

Comment:
This doesn't look right. In the case `src_path` is root `/` and `dest_path` is `/any`, this check is evaluated to `0`, but it should throw.

```suggestion
      dest_path.has_parent_path() &&
      dest_path.parent_path() == src_path) {
```

Reply:
Perhaps 
```
dest_path.parent_path().string().starts_with(src_path.string())
```

Reply:
I'll take a look later today - Sorry for the delay.